editor/py/serve.py

```diff
@@ -568,29 +565,39 @@ def prevent_caching(response):
     return response
 
 
-def get_dataset(file_name):
-    """Reads a .pbtxt file from the user directory and parses it."""
-    with lock(file_name):
-        path = get_path(file_name)
-        if not os.path.isfile(path):
+def get_dataset(name):
+    """Reads a pbtxt proto from the datasets table and parses it."""
+    pbtxt = get_pbtxt(name)
+    dataset = dataset_pb2.Dataset()
+    text_format.Parse(pbtxt, dataset)
+    return dataset
+
+
+def get_pbtxt(name):
+    """Reads a pbtxt proto from the datasets."""
+    with flask.g.db.cursor() as cursor:
+        query = psycopg2.sql.SQL(
+            'SELECT pbtxt FROM datasets WHERE user_id=%s AND dataset_name=%s')
```

Comment:
Are we storing this as bytes now? That will avoid some of the encoding/decoding round trips.

Reply:
Nope, we are storing strings. I agree we should store protos in binary encoding. I expect user-perceptible latency improvements.